warehouse/core/src/main/java/datawave/util/threads/DatawaveUncaughtExceptionHandler.java

```diff
@@ -0,0 +1,55 @@
+package datawave.util.threads;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DatawaveUncaughtExceptionHandler implements Thread.UncaughtExceptionHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(DatawaveUncaughtExceptionHandler.class);
+
+    private static boolean isError(Throwable t, int depth) {
+
+        if (depth > 32) {
+            // This is a peculiar exception. No error has been found, but recursing too deep may cause a
+            // stack overflow so going to stop. Err on the side of caution and assume there could be an
+            // error since not everything was checked.
+            return true;
+        }
+
+        while (t != null) {
+            if (t instanceof Error) {
+                return true;
+            }
+
+            for (Throwable suppressed : t.getSuppressed()) {
+                if (isError(suppressed, depth + 1)) {
+                    return true;
+                }
+            }
+
+            t = t.getCause();
+        }
+
+        return false;
+    }
+
+    static boolean isError(Throwable t) {
+        return isError(t, 0);
+    }
+
+    @Override
+    public void uncaughtException(Thread t, Throwable e) {
+        if (isError(e)) {
+            try {
+                e.printStackTrace();
+                System.err.println("Error thrown in thread: " + t + ", halting VM.");
```

Comment:
Just curious, would it be worthwhile to print this to LOG.error as well to ensure it gets picked up in logging?

Reply:
Agree! Just to be explicit - We should drop the printStackTrace() and the System.err and use a `LOG.error("some message", e)`. That'll give us the message and the full stacktrace and let us configure where/if the logging occurs.